### ISSUE TITLE
Deep Learning Binder Results

### ISSUE THREAD

**Issue by okanders** [RosettaCommons/RFdiffusion#94]
Hi, 
I was attempting to design insulin binders of my own (following the example specifications) and then running the backbone through MPNN_FR and AF2 initial guess. I compared a potential binder to a benchmark insulin binder from the supplement material in Improving de novo protein binder design with deep learning, and even though the target template was identical, I found confusing results:

**InsulinR_mb:**
{'plddt_total': 95.02760208110635, 'plddt_binder': 91.0824370734358, 'plddt_target': 96.7371735844303, 'pae_binder': 2.7015252, 'pae_target': 2.4785695, 'pae_interaction': 4.80579948425293, 'time': 146.24850199604407}

**design_ppi_scaffolded_6_dldesign_4:**
{'plddt_total': 52.28481075101907, 'plddt_binder': 94.86918808372161, 'plddt_target': 33.83158057351464, 'pae_binder': 1.7483437, 'pae_target': 19.690062, 'pae_interaction': 26.69976043701172, 'time': 16.96811721706763}

I am curious as to how you would read these outputs, as shouldn't the plddt of the target both be relatively high given the same template is in use? Is this supposed to indicate that my binder, when recapitulates, fails and impacts the accuracy of the target structure? Thanks and would appreciate the help!

**Comment by nrbennet** [comment]
Can you send the output structure from RFdiffusion and the output structure from MPNN-FR?

**Comment by okanders** [comment]
@nrbennet,
Hi, 
The pdb I sent was the output of the MPNN (I just MPNN used). Here are my outputs from RFDiffusion and the AF2 initial guess from pdb_interfaceAF2.py:


[InsulinDesign_RFDiff_a2pred.zip](https://github.com/RosettaCommons/RFdiffusion/files/12163917/InsulinDesign_RFDiff_a2pred.zip)

Thanks so much for the help!

**Comment by nrbennet** [comment]
Can you send the command you used to generate the diffusion designs?

**Comment by okanders** [comment]
@nrbennet, yes of course. I made some edits to the dl_interface_design.py, but none that should have effected the target binder sequence.

**RFDiffusion (example provided as script):**
../scripts/run_inference.py scaffoldguided.target_path=input_pdbs/insulin_target.pdb inference.output_prefix=example_outputs/ppi_test/design_ppi_scaffolded scaffoldguided.scaffoldguided=True 'ppi.hotspot_res=[A59,A83,A91]' scaffoldguided.target_pdb=True scaffoldguided.target_ss=target_folds/insulin_target_ss.pt scaffoldguided.target_adj=target_folds/insulin_target_adj.pt scaffoldguided.scaffold_dir=./ppi_scaffolds/ inference.num_designs=10 denoiser.noise_scale_ca=0 denoiser.noise_scale_frame=0


**MPNN:**
python dl_interface_design.py -pdbdir ../input/proteinmpnn/insulin -outpdbdir ../output/proteinmpnn/insulin -seqs_per_struct 5
I edited the dl_interface_design file to take in a pdb, the pdb construction at the end seems fine.

**AF2**
python pdb_interfaceAF2predict.py -pdb_dir ../output/proteinmpnn/insulin
I use the pdb interface, as I am not using silent files.

**Comment by nrbennet** [comment]
The outputs of RFdiffusion look fine but they are getting messed up in MPNN. Can you try using the latest version of the `dl_binder_design` scripts? I have modified them to where they can handle silent files or pdb files so you can just use the scripts out of the box without modifying them.

**Comment by okanders** [comment]
@nrbennet 
Unfortunately, I do not have access to pyrosetta... I am trying to look into a Biopython substitute for this MPNN sequencing if that is possible? Thanks so much and truly appreciate all the help!

**Comment by nrbennet** [comment]
It's possible to write the scripts in Biopython, it will just require some work. You will need to go through the script and do all of the chain parsing with Biopython. This is for sure the issue with your results though.

If you're an academic you should be able to get a free license to PyRosetta and this would save you quite a bit of time since you can just use all of the tools out of the box

**Comment by okanders** [comment]
@nrbennet 
Got it, thanks for the help! I might try and convert it for fun, in the mean time as I apply for the license. Would you mind sending me what your post-MPNN looked like so I could benchmark? Thanks again for all the great feedback; such a great tool you made!

**Comment by okanders** [comment]
@nrbennet ,

Hi Nathaniel, I was wondering what your PDB file looks like after ProteinMPNN, as mine is just backbone + encoded sequence, which does not seem to be enough to display an accurate target?

**Comment by nrbennet** [comment]
[mpnn_example.zip](https://github.com/RosettaCommons/RFdiffusion/files/12176843/mpnn_example.zip)

Here is an example of what the structure looks like after MPNN. The script does an automatic cycle of Rosetta FastRelax so the side chains will not be clashing here.
